### PR TITLE
V1.4 find scheduler

### LINKED ISSUE #544: Code: Implement Find command for Scheduler

### PR DESCRIPTION
**Pull request by MatthiasLHK**
fixes #544

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/schedulercommands/FindEventCommand.java [hunk 1]
```diff
@@ -1,17 +1,45 @@
...
 import seedu.address.model.Model;
+import seedu.address.model.event.Event;
+import seedu.address.model.event.FindEventCriteria;
 
 public class FindEventCommand extends Command {
 
     public static final String COMMAND_WORD = "findevent";
 
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all eventd whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
```
Comment (michael-setia) [inline src/main/java/seedu/address/logic/commands/schedulercommands/FindEventCommand.java:21]: Typo in "eventd"

File: src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.address.model.tag.Tag;
+
+public class EventContainsTagsPredicate implements Predicate<Event> {
+    /**
+     * Tags provided by the user to search for contacts containing a matching tag.
+     */
```
Comment (michael-setia) [inline src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java:13]: I think "Contacts" should be "Events"?

File: src/main/java/seedu/address/model/event/FindEventCriteria.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package seedu.address.model.event;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class FindEventCriteria {
+    /**
+     * List of predicates to be used to test for matching contacts.
+     */
```
Comment (michael-setia) [inline src/main/java/seedu/address/model/event/FindEventCriteria.java:10]: Same issue as previous comments.

File: src/main/java/seedu/address/model/event/FindEventCriteria.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+     * @param predicate Predicate to be added.
+     */
+    public void addPredicate(Predicate<Event> predicate) {
+        predicateList.add(predicate);
+    }
+
+    /**
+     * Composes all the predicates in the predicateList field into one predicate to
+     * be used to test for matching contacts.
+     *
```
Comment (michael-setia) [inline src/main/java/seedu/address/model/event/FindEventCriteria.java:25]: Same issue as previous comments.

File: src/main/java/seedu/address/logic/parser/schedulerparsers/FindEventParser.java [hunk 5]
```diff
@@ -1,12 +1,57 @@
...
+        if (argMultiMap.getValue(PREFIX_DATE).isPresent()) {
+            LocalDateTime toSearch = ParserUtil.parseDate(argMultiMap.getValue(PREFIX_DATE).get());
+            List<String> keywordsAsList = List.of(toSearch.toString());
+            EventContainsDatePredicate datePredicate = new EventContainsDatePredicate(keywordsAsList);
+            criteria.addPredicate(datePredicate);
+        }
+        if (argMultiMap.getValue(PREFIX_TAG).isPresent()) {
+            List<String> tags = ParserUtil.parseSearchKeywords(argMultiMap.getValue(PREFIX_TAG).get());
+            Set<Tag> taskTags = ParserUtil.parseTags(tags);
+            EventContainsTagsPredicate tagsPredicate = new EventContainsTagsPredicate(taskTags);
```
Comment (michael-setia) [inline src/main/java/seedu/address/logic/parser/schedulerparsers/FindEventParser.java:48]: I think "taskTags" should be "eventTags"?

File: src/main/java/seedu/address/model/event/EventContainsDatePredicate.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    private final List<String> dates;
+
+    /**
+     * Represents the Predicate for dates.
+     * @param dates date to search for.
+     */
+    public EventContainsDatePredicate(List<String> dates) {
+        this.dates = dates;
+        System.out.println(dates);
+    }
```
Comment (michael-setia) [inline src/main/java/seedu/address/model/event/EventContainsDatePredicate.java:18]: Is `System.out.println(dates);` still being used for debugging? If not maybe we can remove it 😅

File: src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public boolean test(Event event) {
+        requireNonNull(event);
+        boolean isEventTagPresent = !event.getTags().isEmpty();
+        if (isEventTagPresent) {
+            Set<Tag> taskTags = event.getTags();
+            return searchTags.stream()
+                    .anyMatch(tag -> taskTags.contains(tag));
```
Comment (michael-setia) [inline src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java:32]: Same issue as previous comments.

**Review by michael-setia — APPROVED** [review]
LGTM aside from some minor typos (feel free to update in this or future PR).

File: src/main/java/seedu/address/logic/commands/schedulercommands/FindEventCommand.java (hunk 1 above)
Comment (MatthiasLHK) [reply src/main/java/seedu/address/logic/commands/schedulercommands/FindEventCommand.java]: Yup fixed

File: src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java (hunk 2 above)
Comment (MatthiasLHK) [reply src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java]: Yup fixed

File: src/main/java/seedu/address/model/event/FindEventCriteria.java (hunk 3 above)
Comment (MatthiasLHK) [reply src/main/java/seedu/address/model/event/FindEventCriteria.java]: Yup fixed

File: src/main/java/seedu/address/model/event/FindEventCriteria.java (hunk 4 above)
Comment (MatthiasLHK) [reply src/main/java/seedu/address/model/event/FindEventCriteria.java]: Yup fixed

File: src/main/java/seedu/address/logic/parser/schedulerparsers/FindEventParser.java (hunk 5 above)
Comment (MatthiasLHK) [reply src/main/java/seedu/address/logic/parser/schedulerparsers/FindEventParser.java]: Yup fixed

File: src/main/java/seedu/address/model/event/EventContainsDatePredicate.java (hunk 6 above)
Comment (MatthiasLHK) [reply src/main/java/seedu/address/model/event/EventContainsDatePredicate.java]: Yup fixed

File: src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java (hunk 7 above)
Comment (MatthiasLHK) [reply src/main/java/seedu/address/model/event/EventContainsTagsPredicate.java]: Yup fixed

**Pull request merged**